Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/BXLInvocationAnalyzer.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Text;
+using BuildXL.Analyzers.Core.XLGPlusPlus;
+using BuildXL.Scheduler.Tracing;
+using BuildXL.ToolSupport;
+using BuildXL.Utilities;
+using Google.Protobuf;
+
+namespace BuildXL.Execution.Analyzer
+{
+
+    internal partial class Args
+    {
+        public Analyzer InitializeDominoInvocationAnalyzer()
+        {
+            string inputDirPath = null;
+            string outputFilePath = null;
+
+            foreach (var opt in AnalyzerOptions)
+            {
+                if (opt.Name.Equals("outputFile", StringComparison.OrdinalIgnoreCase) ||
+                   opt.Name.Equals("o", StringComparison.OrdinalIgnoreCase))
+                {
+                    outputFilePath = ParseSingletonPathOption(opt, outputFilePath);
+                }
+                else if (opt.Name.Equals("inputDir", StringComparison.OrdinalIgnoreCase) ||
+                     opt.Name.Equals("i", StringComparison.OrdinalIgnoreCase))
+                {
+                    inputDirPath = ParseSingletonPathOption(opt, inputDirPath);
+                }
+                else
+                {
+                    throw Error("Unknown option for event stats analysis: {0}", opt.Name);
+                }
+            }
+
+
+            if (string.IsNullOrEmpty(inputDirPath))
+            {
+                throw Error("/inputDir parameter is required");
+            }
+
+            if (string.IsNullOrEmpty(outputFilePath))
+            {
+                throw Error("/outputFile parameter is required");
+            }
+
+            return new BXLInvocationAnalyzer(GetAnalysisInput())
+            {
+                InputDirPath = inputDirPath,
+                OutputFilePath = outputFilePath
+            };
+        }
+
+        /// <summary>
+        /// Write the help message when the analyzer is invoked with the /help flag
+        /// </summary>
+        private static void WriteDominoInvocationHelp(HelpWriter writer)
+        {
+            writer.WriteBanner("Domino Invocation \"Analyzer\"");
+            writer.WriteModeOption(nameof(AnalysisMode.BXLInvocationXLG), "Gets and outputs information related to domino invocation events from the database.");
+            writer.WriteOption("inputDir", "Required. The directory to read the RocksDB database from", shortName: "i");
+            writer.WriteOption("outputFile", "Required. The file where to write the results", shortName: "o");
+        }
+    }
+
+    /// <summary>
+    /// Analyzer to examine BXL Invocation events that have been dumped into the db
+    /// </summary>
+    internal sealed class BXLInvocationAnalyzer : Analyzer

```

Comment:
>BXLInvocationAnalyzer [](start = 26, length = 21)

I know you just plumbed through the "DominoInvocation" event but I'd expect this same analyzer to be used for the rest of the events too. I suggest naming it something more generic to reflect this meaning